pkg/vm/vm.go

```diff
@@ -950,6 +955,38 @@ func (v *VM) execute(ctx *Context, op opcode.Opcode, parameter []byte) (err erro
 
 		v.estack.PushItem(stackitem.NewBigInteger(new(big.Int).Sqrt(a)))
 
+	case opcode.MODMUL:
+		modulus := v.estack.Pop().BigInt()
+		if modulus.Sign() == 0 {
+			panic("zero modulus")
+		}
+		x2 := v.estack.Pop().BigInt()
+		x1 := v.estack.Pop().BigInt()
+
+		res := new(big.Int).Mul(x1, x2)
+		v.estack.PushItem(stackitem.NewBigInteger(res.Mod(res, modulus)))
+
+	case opcode.MODPOW:
+		modulus := v.estack.Pop().BigInt()
+		exponent := v.estack.Pop().BigInt()
+		base := v.estack.Pop().BigInt()
+		res := new(big.Int)
+		if exponent.Cmp(bigMinusOne) == 0 {
+			if base.Cmp(bigZero) <= 0 {
+				panic("invalid base")
+			}
+			if modulus.Cmp(bigTwo) < 0 {
+				panic("invalid module")
+			}
+			if res.ModInverse(base, modulus) == nil {
+				panic("base and modulus are not relatively prime")
+			}
+		} else {
+			res.Exp(base, exponent, modulus)
```

Comment:
Don't we need to compare with `nil` here?.

Reply:
Ugh, C# doesn't allow `modulus` to be < -1...

Reply:
Fixed.